homeassistant/components/diagnostics/__init__.py

```diff
@@ -84,24 +97,49 @@ async def get(  # pylint: disable=no-self-use
         self, request: web.Request, d_type: str, d_id: str
     ) -> web.Response:
         """Download diagnostics."""
-        if d_type != "config_entry":
+        if d_type not in D_TYPES:
             return web.Response(status=404)
 
         hass = request.app["hass"]
-        config_entry = hass.config_entries.async_get_entry(d_id)
 
-        if config_entry is None:
-            return web.Response(status=404)
+        if d_type == D_TYPE_CONFIG_ENTRY:
+            config_entry = hass.config_entries.async_get_entry(d_id)
 
-        info = hass.data[DOMAIN].get(config_entry.domain)
+            if config_entry is None:
+                return web.Response(status=404)
 
-        if info is None:
-            return web.Response(status=404)
+            info = hass.data[DOMAIN].get(config_entry.domain)
 
-        if info["config_entry"] is None:
-            return web.Response(status=404)
+            if info is None or info[D_TYPE_CONFIG_ENTRY] is None:
+                return web.Response(status=404)
+
+            data = await info[D_TYPE_CONFIG_ENTRY](hass, config_entry)
+            filename = f"{config_entry.domain}-{config_entry.entry_id}"
```

Comment:
Add a return here instead of an "else" spanning rest of the function

```suggestion
            filename = f"{config_entry.domain}-{config_entry.entry_id}"
            return
```